### PR TITLE
Pzsp2-13 visaluziation

### PR DESCRIPTION
**Pull request by noFrostoo**

### CODE REVIEW

File: services/frontend/src/api.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  },
+  async getUser(token, userID) {
+    return axios.get(`${APISUFFIX}/user/${userID}`, authHeaders(token));
+  },
+
+  async getUsers(token) {
+    return axios.get(`${APISUFFIX}/users`, authHeaders(token));
+  },
+
+  // async getPatient(token, email) {
```
Comment (tejotemku) [inline services/frontend/src/api.js:29]: Code smell if you ask me

File: services/frontend/src/api.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  // },
+
+  // async createPatient(token, patient) {
```
Comment (tejotemku) [inline services/frontend/src/api.js:33]: also a code smell

File: services/frontend/src/components/AnalysisInit.vue [hunk 3]
```diff
@@ -29,11 +29,16 @@ export default {
     }
   },
   methods: {
-    startAnalysis() {
+    async startAnalysis() {
+      await this.$store.dispatch("uploadFile");
+      //this.$store.dispatch("startAnalysis");
```
Comment (tejotemku) [inline services/frontend/src/components/AnalysisInit.vue:34]: don't leave commented out stuff

File: services/frontend/src/components/AnalysisVisualization.vue [hunk 4]
```diff
@@ -1,20 +1,136 @@
 <template>
-  <div>
-    <!-- create graphs here -->
-    <!-- this is a mock -->
-    <img src="mock_files/mock_analysis_graph.png"/>
+  <div v-if="analisStared" class="sector flex-column-items-centered card"> Start Analysis before starting </div>
```
Comment (tejotemku) [inline services/frontend/src/components/AnalysisVisualization.vue:2]: "analysis" non "analis". Probably this is not an isolated incident. I suggest quick Find&Replace, e.g. in vscode you can do it on all files in workspace at once.

Comment (tejotemku) [inline services/frontend/src/components/AnalysisVisualization.vue:2]: "Start Analysis before starting" first of all it a bag english sentence. Second of all we were ment to write UI in polish, because we are creating a product for a polish client.

File: services/frontend/src/components/AnalysisVisualization.vue [hunk 5]
```diff
@@ -1,20 +1,136 @@
...
+
 export default {
   name: 'AnalysisVisualization',
   props: ['analysisData'],
-
+  components: {
+    VChart
+  },
+  provide: {
+    [THEME_KEY]: "dark"
```
Comment (tejotemku) [inline services/frontend/src/components/AnalysisVisualization.vue:68]: The webapp is in lightmode color palete currently, I would suggest changing it to light theme too.

File: services/frontend/src/components/AnalysisVisualization.vue [hunk 6]
```diff
@@ -1,20 +1,136 @@
...
+    getCounts() {
+      return  this.results.reduce(function(s, n) {
+         s[n <= 0.80 ? 'below' : 'above'] += 1;
+        return s;
+      }, { above: 0, below: 0 });
+    },
+    setData() {
+      this.results = this.$store.getters.getAnalysisResult;
+      this.len = this.results.length;
+      this.average = Math.round(this.getAverage() * 100) / 100;
```
Comment (tejotemku) [inline services/frontend/src/components/AnalysisVisualization.vue:110]: There is a better way of limiting a number to 2 decimal places. For floats you can use toFixed(decimal_places) method. But aren't we showing user percentages instead of fractions anyway? In that case this is wrong.

File: services/frontend/src/components/AnalysisVisualization.vue [hunk 7]
```diff
@@ -1,20 +1,136 @@
...
+  },
+  methods: {
+    getAverage() {
+      let sum = this.results.reduce((a, b) => a + b, 0);
+      let avg = (sum / this.results.length) || 0;
+      return avg
+    },
+    getCounts() {
+      return  this.results.reduce(function(s, n) {
+         s[n <= 0.80 ? 'below' : 'above'] += 1;
```
Comment (tejotemku) [inline services/frontend/src/components/AnalysisVisualization.vue:103]: Ok this looks cool, just wanted to praise you for it

File: services/frontend/src/components/SearchPatient.vue [hunk 8]
```diff
@@ -55,6 +55,23 @@
...
+      <tr>
+        <th>Pesel</th>
+        <th>Imię Pacjenta</th>
+        <th>Email</th>
+      </tr>  
+      <tr 
+        v-bind:key="patient.id" 
+        v-for="patient in patientList"
+      >
+        <td>11111</td>
```
Comment (tejotemku) [inline services/frontend/src/components/SearchPatient.vue:69]: Let's make the mock beliveable, PESEL is not 5, but 11 digits long.

File: services/frontend/src/store/index.js [hunk 9]
```diff
@@ -19,8 +20,10 @@ export default new Vuex.Store({
...
...3, 0.0025010406970977783, 0.004980981349945068, 0.06438800692558289, 0.44057172536849976, 0.27556610107421875, 0.011737614870071411, 0.002435743808746338, 0.0014040470123291016, 0.006583809852600098, 0.0760832130908966, 0.00637546181678772, 0.003078550100326538, 0.0021862387657165527, 0.004487425088882446, 0.058743685483932495, 0.0885833203792572, 0.02445504069328308, 0.009481340646743774, 0.0038950443267822266, 0.004348307847976685, 0.0029076039791107178, 0.0015988647937774658, 0.008667916059494019, 0.0028187036514282227, 0.001346677541732788, 0.0009747445583343506, 0.013713091611862183, 0.012329459190368652, 0.003603994846343994, 0.006424993276596069, 0.0014544427394866943, 0.008217662572860718, 0.012168645858764648, 0.002413123846054077, 0.003259599208831787, 0.0016272366046905518, 0.0012287795543670654, 0.00694584846496582, 0.024220138788223267, 0.09544476866722107, 0.02614825963973999, 0.008856624364852905, 0.00785788893699646, 0.0017541050910949707, 0.0029064714908599854, 0.0017959773540496826, 0.0017816722393035889, 0.004655629396438599, 0.23912471532821655, 0.3461966812610626, 0.7516708374023438, 0.7929297685623169, 0.9548741579055786, 0.910999059677124, 0.6750947833061218]
```
Comment (tejotemku) [inline services/frontend/src/store/index.js:26]: This single line of code breaks my chrome tab. Also we are not mocking analysis data anymore so get rid of it.

File: services/frontend/src/store/index.js [hunk 10]
```diff
@@ -36,29 +39,79 @@ export default new Vuex.Store({
...
+    setAnalysisID(state, id) {
+      state.analysis.analysisID = id
     }
   },
   actions: {
+    //! those are initial implementation, change when implementing real feature
     //TODO: change email if needed
     async getPatient(state, email) {
-      let patient = await axios.get('user/'+email) // yeah this doesn't work rn 
+      let patient = await api.getPatient(state.state.user.token, email) // yeah this doesn't work rn
```
Comment (tejotemku) [inline services/frontend/src/store/index.js:54]: "state.state" - this is a bad naming scheme

File: services/frontend/src/store/index.js [hunk 11]
```diff
@@ -36,29 +39,79 @@ export default new Vuex.Store({
...
+    //! those are initial implementation, change when implementing real feature
     //TODO: change email if needed
     async getPatient(state, email) {
-      let patient = await axios.get('user/'+email) // yeah this doesn't work rn 
+      let patient = await api.getPatient(state.state.user.token, email) // yeah this doesn't work rn 
       state.commit("setPatient", patient)
     },
     async createPatient(state, patient) {
-      await axios.post('user', patient)
+      await api.createPatient(state.state.user.token, patient)
```
Comment (tejotemku) [inline services/frontend/src/store/index.js:58]: as above

File: services/frontend/src/store/index.js [hunk 12]
```diff
@@ -36,29 +39,79 @@ export default new Vuex.Store({
...
     },
     async getPatients(state) {
-      let patients = await axios.get("users")
-      state.commit('setPatients', patients)
-    }
+      let patients = await api.getUsers()
+      state.commit('setPatients', patients.data)
+    },
+    async startAnalysis(state){
+      let response = await api.startAnalysis(state.state.user.token ,state.state.analysis.recordingID)
```
Comment (tejotemku) [inline services/frontend/src/store/index.js:66]: as above

File: services/frontend/src/store/index.js [hunk 13]
```diff
@@ -36,29 +39,79 @@ export default new Vuex.Store({
...
+      state.commit("setAnalysisID", response.data)
+    },
+    async getAnalysisResults(state) {
+      let statusCode = 404;
+      let data = [];
+      let response;
+      while (statusCode != 200)
+      {
+        response = await api.getFrames(state.state.user.token, state.state.analysis.analysisID).catch(e => {
```
Comment (tejotemku) [inline services/frontend/src/store/index.js:75]: as above

File: services/frontend/src/store/index.js [hunk 14]
```diff
@@ -36,29 +39,79 @@ export default new Vuex.Store({
...
+      console.log(response.data)
+      for(const element of response.data)
+      {
+        data.push(element.probability)
+      }
+      state.commit("setAnalysisResult", data)
+    },
+    async uploadFile (state) {
+      let formData = new FormData();
+      formData.append('file_in', state.state.analysis.file);
```
Comment (tejotemku) [inline services/frontend/src/store/index.js:92]: as above

File: services/frontend/src/store/index.js [hunk 15]
```diff
@@ -36,29 +39,79 @@ export default new Vuex.Store({
...
+      console.log('hui')
+      let response = await api.uploadFile(state.state.user.token, formData)
```
Comment (tejotemku) [inline services/frontend/src/store/index.js:94]: as above

File: services/frontend/src/store/index.js [hunk 16]
```diff
@@ -36,29 +39,79 @@ export default new Vuex.Store({
...
+      state.commit("setRecordingID", response.data)
+      
+      response = await api.startAnalysis(state.state.user.token ,state.state.analysis.recordingID)
```
Comment (tejotemku) [inline services/frontend/src/store/index.js:97]: as above

File: services/frontend/src/store/user/index.js [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+                    await state.dispatch("actionLogOut");
+                }
+            } catch (err) {
+                state.commit("setLoggedIn", true)
+                await state.dispatch("actionLogOut");
+            }
+        },
+        async actionGetUser(state) {
+            try {
+                const response = await api.getUser(state.state.token, state.state.userID)
```
Comment (tejotemku) [inline services/frontend/src/store/user/index.js:55]: "state.state" this is a bad naming scheme

File: services/frontend/src/store/user/index.js [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+                if (response.data) {
+                    //TODO: SET USER DATA
+                }
+            } catch (error) {
+                await state.dispatch("actionCheckApiError", error);
+            }
+        },
+        async actionCheckLoggedIn(state) {
+            if (!state.state.isLoggedIn) {
```
Comment (tejotemku) [inline services/frontend/src/store/user/index.js:64]: as above

File: services/frontend/src/store/user/index.js [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+                let token = state.state.token;
```
Comment (tejotemku) [inline services/frontend/src/store/user/index.js:65]: as above

**Review by tejotemku — COMMENTED** [review]
All in all it works, but there are some things that should be corrected

File: services/frontend/src/components/AnalysisVisualization.vue (hunk 7 above)
Comment (noFrostoo) [reply services/frontend/src/components/AnalysisVisualization.vue:103]: thanks

File: services/frontend/src/store/index.js (hunk 11 above)
Comment (noFrostoo) [reply services/frontend/src/store/index.js:57]: done, changed to contex.state

File: services/frontend/src/store/index.js (hunk 10 above)
Comment (noFrostoo) [reply services/frontend/src/store/index.js:53]: done, changed to contex.state

File: services/frontend/src/store/index.js (hunk 12 above)
Comment (noFrostoo) [reply services/frontend/src/store/index.js:65]: done, changed to contex.state

File: services/frontend/src/store/index.js (hunk 13 above)
Comment (noFrostoo) [reply services/frontend/src/store/index.js:74]: done, changed to contex.state

File: services/frontend/src/store/index.js (hunk 14 above)
Comment (noFrostoo) [reply services/frontend/src/store/index.js:91]: done, changed to contex.state

File: services/frontend/src/store/index.js (hunk 15 above)
Comment (noFrostoo) [reply services/frontend/src/store/index.js:93]: done, changed to contex.state

File: services/frontend/src/store/index.js (hunk 16 above)
Comment (noFrostoo) [reply services/frontend/src/store/index.js:96]: done, changed to contex.state

File: services/frontend/src/store/user/index.js (hunk 17 above)
Comment (noFrostoo) [reply services/frontend/src/store/user/index.js:55]: done, changed to contex.state

File: services/frontend/src/store/user/index.js (hunk 19 above)
Comment (noFrostoo) [reply services/frontend/src/store/user/index.js:65]: done, changed to contex.state

File: services/frontend/src/store/user/index.js (hunk 18 above)
Comment (noFrostoo) [reply services/frontend/src/store/user/index.js:64]: done, changed to contex.state

File: services/frontend/src/components/AnalysisVisualization.vue [hunk 20]
```diff
@@ -1,20 +1,136 @@
 <template>
-  <div>
-    <!-- create graphs here -->
-    <!-- this is a mock -->
-    <img src="mock_files/mock_analysis_graph.png"/>
+  <div v-if="analysisStared" class="sector flex-column-items-centered card"> Rozpocznij analizę </div>
```
Comment (tejotemku) [inline services/frontend/src/components/AnalysisVisualization.vue:2]: I think it should be analysisStarted not analysisStared

File: services/frontend/src/components/AnalysisVisualization.vue [hunk 21]
```diff
@@ -1,20 +1,136 @@
...
+    <div class="flex-row-items-centered" style="margin-block-end: 30px">
+
+      <table style="margin-inline-end: 30px">
+        <tr>
+          <td>Pesel </td>
+          <td>01234567890 </td>
+        </tr>
+        <tr>
+      <td>Imie</td>
+      <td>Daniel</td>
```
Comment (tejotemku) [inline services/frontend/src/components/AnalysisVisualization.vue:15]: If you have time you can already change it to display a variable value. It will save time l8r when we will be connecting it with backend

File: services/frontend/src/components/AnalysisVisualization.vue [hunk 22]
```diff
@@ -1,20 +1,136 @@
...
+        </tr>
+        <tr>
+            <td>Nazwisko</td>
+      <td>Lipniacki </td>
```
Comment (tejotemku) [inline services/frontend/src/components/AnalysisVisualization.vue:19]: As above

File: services/frontend/src/components/AnalysisVisualization.vue [hunk 23]
```diff
@@ -1,20 +1,136 @@
...
+  <div v-if="analysisStared" class="sector flex-column-items-centered card"> Rozpocznij analizę </div>
+
+  <div v-else class="sector flex-column-items-centered card">
+
+    <div class="flex-row-items-centered" style="margin-block-end: 30px">
+
+      <table style="margin-inline-end: 30px">
+        <tr>
+          <td>Pesel </td>
+          <td>01234567890 </td>
```
Comment (tejotemku) [inline services/frontend/src/components/AnalysisVisualization.vue:11]: As below xD

File: services/frontend/src/components/AnalysisVisualization.vue [hunk 24]
```diff
@@ -1,20 +1,136 @@
...
+
 export default {
   name: 'AnalysisVisualization',
   props: ['analysisData'],
-
+  components: {
+    VChart
+  },
+  provide: {
+    [THEME_KEY]: "vintage"
```
Comment (tejotemku) [inline services/frontend/src/components/AnalysisVisualization.vue:68]: Is there a retro option?

File: services/frontend/src/components/SignIn.vue [hunk 25]
```diff
@@ -19,7 +19,7 @@
     />
     <a  
       class="input-element-standard button link"
-      href="https://www.google.com"
+      href="https://www.w3schools.com"
```
Comment (tejotemku) [inline services/frontend/src/components/SignIn.vue:22]: Just out of curiosity, why did you change that?

File: services/frontend/src/store/index.js [hunk 26]
```diff
@@ -9,56 +10,116 @@ Vue.use(Vuex)
...
+  analysis: {
+    recordingID: null,
+    analysisID: null,
+    file: null,
+    result: null,
+  },
+  patients: []
+}
+
+export const mutations = {
```
Comment (tejotemku) [inline services/frontend/src/store/index.js:30]: I am not sure if it is a good name, why not just setters? You have gettters below

**Review by tejotemku — COMMENTED** [review]
Mainly changes related to good programming practices

File: services/frontend/src/store/index.js (hunk 26 above)
Comment (noFrostoo) [reply services/frontend/src/store/index.js:30]: this name comes from vuex, and is standat vuex naming. not sure we can even change it

File: services/frontend/src/components/SignIn.vue (hunk 25 above)
Comment (noFrostoo) [reply services/frontend/src/components/SignIn.vue:22]: privacybadger said that this was a tracker, and i want to have no trackes for now :D

File: services/frontend/src/components/AnalysisVisualization.vue (hunk 24 above)
Comment (noFrostoo) [reply services/frontend/src/components/AnalysisVisualization.vue:68]: yeah, it should have colors matching ours but it only changes of line on the chart

File: services/frontend/src/components/AnalysisVisualization.vue (hunk 21 above)
Comment (noFrostoo) [reply services/frontend/src/components/AnalysisVisualization.vue:15]: i think this should be done in different PR, but i have time and can do it now

**Review by tejotemku — APPROVED** [review]
